numba/np/arraymath.py

```diff
@@ -791,6 +792,7 @@ def flat_all(a):
 @overload_method(types.Array, "any")
 def np_any(a):
     def flat_any(a):
+        a = np.asarray(a)
```

Comment:
Would be good to add a test for this, perhaps with a list as per the original bug?

Reply:
Sure,  I'll add a test case. Thanks